src/test/java/com/ericsson/ei/subscriptionhandler/test/SubscriptionValidatorTest.java

```diff
@@ -410,35 +426,52 @@ public void validateNotificationTypeInvalidTypeTest() throws Exception {
         assertTrue(false);
     }
 
-	
-	// TODO: Validator Unit tests for Jmepath syntax validator needs to be implemented here.
-	
-	
-	/**
-	 * Unit tests for testing a whole  Subscription Json object.
-	 * Valid Result: true or false
-	 */
-	@Test
-	public void validateFullSubscriptionWithValidSubscriptionParameters() {
-		try {
-			subscriptionValidator.validateSubscription(subscriptionValid);
-		}
-		catch (SubscriptionValidationException e) {
-			assertTrue(e.getMessage(), false);
-			return;
-		}
-		assertTrue(true);
-	}
-	
-	@Test
-	public void validateFullSubscriptionWithInvalidSubscriptionParameters() {
-		try {
-			subscriptionValidator.validateSubscription(subscriptionInvalid);
-		}
-		catch (SubscriptionValidationException e) {
-			assertTrue(true);
-			return;
-		}
-		assertTrue(false);
-	}
+    // TODO: Validator Unit tests for Jmepath syntax validator needs to be
+    // implemented here.
+
+    /**
+     * Unit tests for testing a whole Subscription Json object. Valid Result: true
+     * or false
+     */
+    @Test
+    public void validateFullSubscriptionWithValidSubscriptionParameters() {
+        try {
+            subscriptionValidator.validateSubscription(subscriptionValid);
+        } catch (SubscriptionValidationException e) {
+            assertTrue(e.getMessage(), false);
+            return;
+        }
+        assertTrue(true);
+    }
+
+    @Test
+    public void validateFullSubscriptionWithInvalidSubscriptionParameters() {
+        try {
+            subscriptionValidator.validateSubscription(subscriptionInvalid);
+        } catch (SubscriptionValidationException e) {
+            assertTrue(true);
+            return;
+        }
+        assertTrue(false);
+    }
+
+    @Test
+    public void validateSubscriptionWithSchemaTest() throws Exception {
+        subscriptionValid.setSubscriptionName("myName");
+        try {
+            invokeMethod(subscriptionValidator, "validateWithSchema", subscriptionValid);
+        } catch (SubscriptionValidationException e) {
+            assertTrue(false);
+        }
+
+        subscriptionValid.setSubscriptionName(null);
+        try {
+            invokeMethod(subscriptionValidator, "validateWithSchema", subscriptionValid);
+        } catch (SubscriptionValidationException e) {
+            assertTrue(true);
```

Comment:
assertTrue(true) makes no sense. Removing it will have the same effect so please remove all assertTrue(true).

Reply:
done